common/client-core/src/init/helpers.rs

```diff
@@ -133,24 +145,34 @@ impl<'a, G: ConnectableGateway> GatewayWithLatency<'a, G> {
 }
 
 pub async fn gateways_for_init<R: Rng>(
-    rng: &mut R,
+    _rng: &mut R,
     nym_apis: &[Url],
     user_agent: Option<UserAgent>,
     minimum_performance: u8,
     ignore_epoch_roles: bool,
 ) -> Result<Vec<RoutingNode>, ClientCoreError> {
-    let nym_api = nym_apis
-        .choose(rng)
-        .ok_or(ClientCoreError::ListOfNymApisIsEmpty)?;
+    // Build client with ALL URLs for fallback support
+    let nym_api_urls: Vec<nym_http_api_client::Url> = nym_apis
+        .iter()
+        .filter_map(|url| nym_http_api_client::Url::parse(url.as_str()).ok())
+        .collect();
+
+    if nym_api_urls.is_empty() {
+        return Err(ClientCoreError::ListOfNymApisIsEmpty);
+    }
 
-    // Use the unified HTTP client directly with optional user agent
-    let mut builder = nym_http_api_client::Client::builder(nym_api.clone())
-        .map_err(|e| {
+    let mut builder = if nym_api_urls.len() == 1 {
+        nym_http_api_client::Client::builder(nym_api_urls[0].clone()).map_err(|e| {
             ClientCoreError::ValidatorClientError(nym_validator_client::ValidatorClientError::from(
                 e,
             ))
         })?
-        .with_bincode(); // Use bincode for better performance
+    } else {
+        nym_http_api_client::ClientBuilder::new_with_urls(nym_api_urls.clone()).with_retries(3)
+        // Enable URL rotation on failure
+    };
+
+    builder = builder.with_bincode();
```

Comment:
```suggestion
    let builder = nym_http_api_client::ClientBuilder::new_with_urls(nym_api_urls.clone())
      .with_retries(3)
      .with_bincode();
```

as a side note, it's a bit suspicious that constructing with one url could return an error, while using multiple is seemingly infallible...

Reply:
also, but that's a nit, perhaps make number of retries configurable